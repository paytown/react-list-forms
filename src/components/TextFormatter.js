import React, { PureComponent } from 'react';

export default class TextFormatter extends PureComponent {
  state = {
    text: '',
    formattedText: '',
    color: '#ff0000'
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const {
      text,
      color
    } = this.state;

    return (
      <>
      <input type="text" name="text" value={text} onChange={this.handleChange} />
      <input type="color" name="color" value={color} onChange={this.handleChange} />
      <pre style={{ color }}>{text}</pre>
      </>
    );
  }
}
