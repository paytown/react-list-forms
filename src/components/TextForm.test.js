import React from 'react';
import { shallow } from 'enzyme';
import TextFormatter from './TextFormatter';

describe('text formatter', () => {
  it('change the format of text input', () => {
    const wrapper = shallow(<TextFormatter />);
    expect(wrapper).toMatchSnapshot();
  });
});
