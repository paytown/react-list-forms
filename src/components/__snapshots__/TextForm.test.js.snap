// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text formatter change the format of text input 1`] = `
<Fragment>
  <input
    name="text"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="#ff0000"
  />
  <pre
    style={
      Object {
        "color": "#ff0000",
      }
    }
  />
</Fragment>
`;
